### PR TITLE
WIP: JN30B

### PR DESCRIPTION
**Pull request by wremie**
Adding support for the Auvidea JN30B carrier board;

- added .coffee and .svg for jn30b
- conf/machine: added machine configuration
- added .dtb from [Auvidea firmware](https://auvidea.eu/firmware/)
- .bbappend: SRC_URI and do_deploy_append

Image builds (-m jn30b-nano) but CSI camera is not showing up as node in /dev/video*. Most probably I forget about some things as I am quite new to Yocto. Firmware works when applied to NVIDIA JetPack. Somebody willing to review and push me in the right direction?

### CODE REVIEW

**Comment by wremie** [comment]
Changing the .dtb file seems to have fixed the issue. Camera nodes are now popping up in /dev/video*.

Auvidea is supplying the sources with firmware. Therefore, patching the kernel seems to be better IMO. However, I'm no expert so can the admins please advice on best practice?

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 1]
```diff
@@ -185,3 +189,7 @@ do_deploy_append_srd3-tx2() {
 do_deploy_append_blackboard-tx2() {
     cp ${WORKDIR}/tegra186-tx2-blackboard.dtb "${DEPLOYDIR}"
 }
+
+do_deploy_append_jn30b-nano() {
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: As now Jetson Nano splits into two separate devices,  it's necessary to use different dtbs for tegra210-flash too, like we did here for the tx2: https://github.com/balena-os/balena-jetson/commit/fce0477aa41c74c2698a56077b15bfb9e563e609

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 2]
```diff
@@ -34,7 +34,7 @@ SRC_URI_append_jetson-tx2 = " \
 "
 
 SRC_URI_append_jetson-nano = " \
-    file://tegra210-p3448-0002-p3449-0000-b00.dtb \
+    file://tegra210-p3448-0000-p3449-0000-a02.dtb \
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: This seems to override the default device tree for jetson-nano and swap it to the JN30B one. Think it's best to only add it for JN30B,  with its original name, maybe append a JN30B_JP4.2.2 to it.. This name will be specified in tegra210-flash, as mentioned in my other comment. Should make it easier to maintain in the future. Thanks

Comment (wremie) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Makes sense, I've renamed the .dtb file name to `tegra210-p3448-0002-p3449-0000-b00-jn30b-JP4.2.2.dtb`

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 1 above)
Comment (wremie) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: Made the split within tegra210-flash by including KERNEL_DEVICETREE_jn30b-nano

**Comment by wremie** [comment]
@acostach, thanks for your feedback. I've processed your suggestions which introduced a new bug relating the .dtb encryption. RESIN_BOOT_PARTITION_FILES still seems to reference to the Jetson-nano .dtb which is not available.

`ERROR: resin-image-1.0-r0 do_rootfs: /home/ubuntu/balena-jetson/build/tmp/deploy/images/jn30b-nano/bootfiles/tegra210-p3448-0000-p3449-0000-a02.dtb.encrypt is an invalid path referenced in RESIN_BOOT_PARTITION_FILES.`

`ERROR: resin-image-1.0-r0 do_rootfs: Function failed: resin_boot_dirgen_and_deploy (log file is located at /home/ubuntu/balena-jetson/build/tmp/work/jn30b_nano-poky-linux/resin-image/1.0-r0/temp/log.do_rootfs.10507)`

**Comment by acostach** [comment]
@wremie Can you please try by specifying the device tree in the machine conf - jn30b-tx2.conf - that you added? 

That name is being used to place the encrypted dtb in the boot partition as seen here: https://github.com/balena-os/balena-jetson/blob/8833a908f6d35510112ad297b4171c85e2950876/layers/meta-balena-jetson/recipes-core/images/resin-image.inc#L7

Thanks

**Comment by wremie** [comment]
@acostach, thanks for the fast response. Just to understand, defining the KERNEL_DEVICETREE in .conf is needed because of the global scope right? Your earlier suggestion to specify the KERNEL_DEVICETREE in tegra210-flash is local to the recipe only?

According to following [docs](https://www.yoctoproject.org/docs/2.5/ref-manual/ref-manual.html#var-KERNEL_DEVICETREE) I've specified the filename. By doing so encryption seems to work but the compile task of linux-tegra is failing complaining about a missing rule for the .dtb file [log.do_compile.20090.txt](https://github.com/balena-os/balena-jetson/files/3775779/log.do_compile.20090.txt).

**Comment by acostach** [comment]
@wremie yes, you're right, we're using the prebuilt dtb in this case, and that will trigger build in the kernel. please switch the KERNEL_DEVICETREE from machine.conf to appending
`DTBFILE_jn30b-nano = "tegra210-p3448-0002-p3449-0000-b00-jn30b-JP4.2.2.dtb"`
in resin-image.inc

Also, if you would like to add this in balena-jetson, please create a PR that adds commit starting from current HEAD and is targeted against master. Please also re-word all the commit messages to follow the standard commit guidelines, i.e:

```
resin-image: Specify DTB for Auvidea JN30b

Commit message goes here

Changelog-entry: resin-image: Specify DTB for Auvidea JN30b
Signed-off-by: Name <address@mail.com
```

Changelog-entry is necessary on only 1 commit for each PR, and commits need to be split by packages, i.e: machine, resin-image, linux-tegra, tegra210-flash, etc

Thank you

**Comment by acostach** [comment]
Thank you for contributing this board @wremie , I've re-grouped and formatted the commits to follow the commit guidelines in PR https://github.com/balena-os/balena-jetson/pull/26 against master branch.

**Pull request merged**